examples/CalcQt/CMakeLists.txt

```diff
@@ -1,36 +1,20 @@
 project(CalcQt)
 
 set(CALCQT_HEADERS src/CalculatorWidget.h)
-set(CALCQT_SOURCES src/CalcQt.cpp src/CalculatorWidget.cpp)
-include_directories(${CUKE_INCLUDE_DIRS} src)
+include_directories(src)
 
-find_package(Qt5Core QUIET)
-find_package(Qt5Widgets QUIET)
-find_package(Qt5Test QUIET)
-
-if(${Qt5Core_FOUND} AND ${Qt5Widgets_FOUND} AND ${Qt5Test_FOUND})
-    set(CMAKE_INCLUDE_CURRENT_DIR ON)
-    set(CMAKE_AUTOMOC ON)
-    set(QT_LIBRARIES Qt5::Core Qt5::Widgets Qt5::Test)
-
-    add_library(libcalcqt src/CalculatorWidget.cpp ${CALCQT_HEADERS})
+if(QT_LIBRARIES)
+    if(QT4_FOUND)
+        qt4_wrap_cpp(CALCQT_MOC ${CALCQT_HEADERS})
+    else()
+        qt5_wrap_cpp(CALCQT_MOC ${CALCQT_HEADERS})
+    endif()
+    add_library(libcalcqt src/CalculatorWidget.cpp ${CALCQT_MOC})
```

Comment:
I thought you would like such self-contined implementation better. Automoc is fine by me :)